tidy3d/plugins/design/result.py

```diff
@@ -186,8 +188,27 @@ def to_dataframe(self) -> pandas.DataFrame:
             data.append(data_i)
 
         val_keys = list(self.value_as_dict(self.values[0])) if self.values else [""]
+
         columns = list(self.dims) + val_keys
 
+        if include_aux:
+            if self.aux_values is not None:
+                # Can use [0] for aux keys as the function is assumed producing the same structure of output each run
+                if all(isinstance(auxs, dict) for auxs in self.aux_values):
```

Comment:
sort of assumes that all of the `aux_datas[I]` are the same type and if dicts, they have the same keys?

Reply:
This is true, though I think instances where this is false are very niche. If people have used `Result.combine` then it's been checked that the source and the columns have passed as the same. Particularly if the source is the same then you'd expect the augs to be identical.

Thinking about it the only way I can think of that breaks this is if people supply `fn_post` with different return statements which could be seen as valid for e.g. pseudocode

```
if sim_data["something"] == None:
    return fit_val, None
else:
    aug_dict = make_aug_dict(sim_data)
    return fit_val, aug_dict
```
Not really sure how to handle this case. It's probably unlikely so I suggest we wait and see if anyone does it. I think the current code could be modified easily enough if people wanted this functionality.

Reply:
> Thinking about it the only way I can think of that breaks this is if people supply fn_post with different return statements which could be seen as valid for e.g. pseudocode

yea I think this is what I had in mind as possibly breaking this code. I think if you dont want to handle it, we just need to just add a quick check that if aux_dict is supplied, that they all are the same type, and if that type is a dict it has the same keys, lists / tuples have the same length, etc. ? otherwise either raise an error that we dont know how to build the result using the aux_data provided or just warn and skip the aux data altogether?

Reply:
I've got a light catch now where  `_extract_output` checks that all outputs have the same `type`.  This will catch crude multi-returns with an error that says raise a github issue if this is effecting you.

It is possible to sneak through an invalid output if the say the user has a return of `[float, list]` and `[float, dict]` but this will only fail when trying to create a dataframe. We're getting into the weeds with edge cases at this point, and it would be an easy enough patch to include this functionality if required.